flyway-core/src/main/java/org/flywaydb/core/internal/dbsupport/DryRun.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright 2010-2016 Boxfuse GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.flywaydb.core.internal.dbsupport;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Savepoint;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.WeakHashMap;
+import org.flywaydb.core.api.FlywayException;
+
+public class DryRun {
+    private static final String SAVEPOINT_MARKER = "dryrun";
+    private final Map<Connection, Savepoint> dryRunSavepoints = new WeakHashMap<Connection, Savepoint>();
+    private final int connectionCount;
+    
+    public DryRun(int transactionIsolationLevel, DbSupport dbSupport, Connection... connections) {
+        if (dbSupport == null) {
+            throw new NullPointerException("A Dry-run requires dbSupport to determine whether connections can support a rollback successfully.");
+        }
+
+        if (!dbSupport.supportsDdlTransactions()) {
```

Comment:
@aldas - see validation here... I actually had a block of code which would set up a rollback point and then test rollbacking (catching exceptions) which validated driver that way; however, it looks like Flyway folks had already done this testing on a per db type as supportsDdlTransactions is set on the support object.

If might be good to re-add that validation in addition to the dbSupport value though.  In case a db is configured to not support the transactions or something like that.